compiler/codegen/ArrayInsnsCodeGen.cpp

```
@@ -42,9 +42,11 @@ void NonTerminatorInsnCodeGen::visit(ArrayLoadInsn &obj, llvm::IRBuilder<> &buil
 
     auto *lhsOpRef = functionGenerator.getLocalOrGlobalVal(obj.lhsOp);
     auto *rhsOpTempRef = functionGenerator.createTempVal(obj.rhsOp, builder);
+    auto *bitCastRhsOpTempRef = builder.CreateBitCast(
+        rhsOpTempRef, builder.getInt8PtrTy());
```

Make a note that this is also only a temp requirement as we can update the load and store function declaration to pass a DynamicBalArray type pointer.

---

Changed the function definition to use the new type and add the comment there as well saying for other array types it is silently ignored